src/coprocessor/codec/mysql/json/json_cast.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::Json;
+use super::Result;
+
+impl Json {
+    pub fn cast_to_int(self) -> Result<i64> {
```

Comment:
I think if other `cast_to_*` functions return `Result`, it's not a bad idea to keep similar with them.